### PR TITLE
Govuk feedback processing

### PR DESCRIPTION
**Pull request by j-marvin-GDS**
The objective of this PR is to implement changes to the documentation for the GOV.UK Feedback Processing project in the community technical documentation. 

Those changes include: 

1. **The removal of existing documentation**: the previous feedback processing documentation was kept under the 'Reporting' section of the documentation, and was largely out of date given recent developments to the feedback processing tooling. It is proposed that the **existing documentation is deleted** and the **reporting section is removed**
2. **The creation of new documentation**: new feedback processing documentation will be created in the engineering section; this documentation details all of the components of the GOV.UK feedback processing tooling, including inputs, outputs, and execution frequency

The objective of the new documentation is to document, for each component of the GOV.UK feedback processing tooling: 

- What the component is used for
- What the key configuration values are for the component
- For executable components, what the relevant inputs, outputs, and processing times are
- What technical dependencies the component has
- How the component should be used (if relevant)

A concern to be addressed during the review is whether there is any potential sensitivity to the contents of the documentation bearing in mind that when published it will be publicly accessible. The documentation references details relating to network security and contains names of globally-unique storage buckets - is this a problem?

### CODE REVIEW

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+pages.
+
+The objective of the **GOV.UK Feedback Processing** project is to facilitate the analysis of user feedback data by: 
+
+1. Consolidating feedback data from multiple sources into a single source/format
+2. Sanitising feedback data by removing or masking potentially offensive, private, or sensitive text
+
+The project meets these objectives by: 
+
+1. Delivering technical components enabling the sourcing, consolidation and sanitisation of user feedback data
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:21]: comma missing?

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+2. Defining 'workflows' which orchestrate the use of technical components into an executable process
+3. Implementing the mechanisms used to automate the execution of workflows (daily)
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:23]: a complex sentence (but maybe that's ok)

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+The BigQuery table uses [field nesting](https://cloud.google.com/bigquery/docs/best-practices-performance-nested) to 
+capture all prompts and their associated responses against a single record for a given user submission. 
+
+The table is [partitioned](https://cloud.google.com/bigquery/docs/partitioned-tables) by feedback record creation date.
+Queries against the table **must** contain a filter on the field containing the record creation date (`created`).
+
+- [Resource definition](https://github.com/alphagov/govuk-feedback-processing/blob/main/terraform/main.tf#L16)
+- [Schema definition](https://github.com/alphagov/govuk-feedback-processing/blob/main/terraform/feedback_data_schema.json)
+
+## Networking Componentry
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:54]: I learnt a new word! :)

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+# Components
+
+## BigQuery Table
+
+The BigQuery table to which processed data is saved. 
+
+A single user feedback submission may contain multiple user 'responses' where a user has been issued more than one 
+'prompt' for feedback or comment.
+
+The BigQuery table uses [field nesting](https://cloud.google.com/bigquery/docs/best-practices-performance-nested) to
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:45]: it's useful to link to google docs for this kind of thing! thanks

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+- [Schema definition](https://github.com/alphagov/govuk-feedback-processing/blob/main/terraform/feedback_data_schema.json)
+
+## Networking Componentry
+
+It is necessary to create 'Virtual Private Cloud' (VPC) in order to enable secure communication between components 
+(such as between a Cloud Function/API and a Cloud SQL instance).
+
+The following network components have been created: 
+
+- A [compute subnetwork](https://cloud.google.com/vpc/docs/subnets) - [definition](https://github.com/alphagov/govuk-feedback-processing/blob/main/terraform/main.tf#L31)
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:61]: excellent documentation! takes me to the correct line I assume?

Might be worth stating how one gets access to this repo at the start of the section, currently I'm 404'ing

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+1. Fetch the latest Support API Datastore backup file from the S3 Mirror Cloud Storage bucket
+2. Restore the Support API Datastore backup file to a Google Cloud-hosted Postgres instance
+3. Read the `raw` Support API data from the Google Cloud-hosted Postres instance into the staging Cloud Storage bucket
+4. Process the `raw` Support API data from the staging bucket
+5. Save the output records into Google BigQuery.
+
+The workflow calls the [`propogate-support-api`](#propogate-support-api-data), [`read-raw-feedback`](#read-raw-feedback)
+and [`process-raw-feedback`](#process-raw-feedback) cloud functions in sequence. The process is run each time a new
+Support API Datastore backup file is created in the S3 Mirror Cloud Storage bucket; typically, this occurs daily at 
+06:00AM.
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:178]: GMT? i know we have sometimes had issues with timezones, might be worth specifying?

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+The [`better-profanity`](https://pypi.org/project/better-profanity/) Python package is used in combination with a 
+pre-compiled [list of censored words](https://github.com/alphagov/govuk-feedback-processing/blob/main/src/utils/profane_words.txt)
+to: 
+
+1. Identify instances of profanity in user feedback text
+2. Replace identified instances of profanity with a mask value ('****').
+
+## PII Removal 
+
+The objective of PII removal is remove from feedback text any articles of data - such as phone numbers, names, or
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:302]: is to remove

File: source/engineering/govuk-feedback-processing/index.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+The objective of PII removal is remove from feedback text any articles of data - such as phone numbers, names, or 
+National Insurance numbers - which could be used to uniquely identify an individual.
+
+Two methods are used to achieve this: 
+
+1. [Google Cloud DLP](https://cloud.google.com/dlp): Feedback data is parsed using Google's Data Loss Prevention (DLP) 
+   API. The DLP tools can be configured to scan feedback text for 
+   specific [types of PII](https://github.com/alphagov/govuk-feedback-processing/blob/main/src/transforms/utils.py#L10), 
+   and will replace identified instances of PII with an appropriate mask, e.g. ['PERSON_NAME']
+2. Text pattern matching: Feedback data is scanned to identify the presence of text combinations which match
```
Comment (mammykins) [inline source/engineering/govuk-feedback-processing/index.html.md.erb:311]: using regular expressions?

**Review by mammykins — APPROVED** [review]
I read through the documentation on Github. 

Excellent docs!

The docs made good use of links and the interested reader could readily follow up and delve deeper if required. The language was clear and the structure seemed to be effective in breaking up the complexity, however hard to assess until it renders.

I visually inspected hyperlinks but didn't test they worked as 404'd due to lack of access to repo.

Some minor comments and typos.

:shipit:

File: source/engineering/govuk-feedback-processing/index.html.md.erb (hunk 1 above)
Comment (j-marvin-GDS) [reply source/engineering/govuk-feedback-processing/index.html.md.erb:21]: Updated the wording + added a comma

**Pull request merged**